packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -255,6 +270,15 @@ export default class NewVm extends BaseComponent {
     if (get(() => prevProps.template.id) !== get(() => this.props.template.id)) {
       this._initTemplate(this.props.template)
     }
+
+    if (
+      !isEqual(prevProps.resourceSets, this.props.resourceSets) ||
+      prevProps.location.query.resourceSet !== this.props.location.query.resourceSet
```

Comment:
No, because in case you arrive directly on the VM creation page (eg: after page reload or by clicking on a URL: `.../vms/new?resourceSet=<ID>&template`, then we will not initialize the correct value.

Reply:
Why not?

Reply:
Because in the cases previously defined, `NewVm` component will be mounted with `props.location.query.resourceSet` as non undefined value. So `this._setState({share:...})` will not be triggered except if you select another resourceSet.